world_boss/app/api.py

```diff
@@ -118,75 +144,79 @@ def prepare_transfer_assets() -> Response:
+async def next_tx_nonce(
+    request: Request, channel_id: Annotated[str, Form()], db: Session = Depends(get_db)
+):
+    nonce = get_next_tx_nonce(db)
     client.chat_postMessage(
         channel=channel_id,
         text=f"next tx nonce: {nonce}",
     )
-    return jsonify(200)
+    return JSONResponse(200)
 
 
 @api.post("/prepare-reward-assets")
 @slack_auth
-def prepare_reward_assets():
-    channel_id = request.values.get("channel_id")
-    raid_id = request.values.get("text", type=int)
-    task = upload_prepare_reward_assets.delay(channel_id, raid_id)
-    return jsonify({"task_id": task.id})
+async def prepare_reward_assets(
+    request: Request, channel_id: Annotated[str, Form()], text: Annotated[int, Form()]
+):
+    task = upload_prepare_reward_assets.delay(channel_id, text)
+    return JSONResponse(task.id)
 
 
 @api.post("/stage-transaction")
 @slack_auth
-def stage_transactions():
-    channel_id = request.values.get("channel_id")
-    network = request.values.get("text")
+async def stage_transactions(
+    request: Request,
+    channel_id: Annotated[str, Form()],
+    text: Annotated[str, Form()],
+    db: Session = Depends(get_db),
+):
     nonce_list = (
-        db.session.query(Transaction.nonce)
+        db.query(Transaction.nonce)
         .filter_by(signer=signer.address, tx_result=None)
         .all()
     )
     network_type = NetworkType.INTERNAL
-    if network.lower() == "main":
+    if text.lower() == "main":
         network_type = NetworkType.MAIN
     headless_urls = HEADLESS_URLS[network_type]
     task = chord(
         stage_transaction.s(headless_url, nonce)
         for headless_url in headless_urls
         for nonce, in nonce_list
     )(send_slack_message.si(channel_id, f"stage {len(nonce_list)} transactions"))
-    return jsonify({"task_id": task.id})
+    return JSONResponse(task.id)
 
 
 @api.post("/transaction-result")
 @slack_auth
-def transaction_result():
-    channel_id = request.values.get("channel_id")
-    network = request.values.get("text")
-    tx_ids = db.session.query(Transaction.tx_id).filter_by(tx_result=None)
-    network_type = NetworkType.INTERNAL
-    if network.lower() == "main":
-        network_type = NetworkType.MAIN
-    url = MINER_URLS[network_type]
+async def transaction_result(
+    request: Request,
+    channel_id: Annotated[str, Form()],
+    db: Session = Depends(get_db),
+):
+    tx_ids = db.query(Transaction.tx_id).filter_by(tx_result=None)
+    url = MINER_URLS[NetworkType.MAIN]
     task = chord(query_tx_result.s(url, str(tx_id)) for tx_id, in tx_ids)(
         upload_tx_result.s(channel_id)
     )
-    return jsonify({"task_id": task.id})
+    return JSONResponse(task.id)
 
 
 @api.post("/balance")
 @slack_auth
-def check_balance():
-    channel_id = request.values.get("channel_id")
-    currencies = get_currencies()
+async def check_balance(
+    request: Request, channel_id: Annotated[str, Form()], db: Session = Depends(get_db)
```

Comment:
If `request` object it not used inside function, you can remove it from function signature.

Reply:
Hmm.. I think there is better way to deal with it but that's not a big deal at this point. :innocent:

Reply:
I left it for compatibility, but I don't want to use the Slack API anymore when doing https://github.com/planetarium/world-boss-service/issues/44